neuraxle/checkpoints.py

```diff
@@ -43,8 +54,6 @@ def fit(self, data_inputs, expected_outputs=None) -> 'BaseCheckpointStep':
         :param data_inputs: data inputs to save
         :return: self
         """
-        self.save_checkpoint(data_inputs)
```

Comment:
Could we already rename `BaseCheckpointStep` to `BaseDataCheckpoint`, as we'll also have `BaseStepCheckpoint` later ?

Also, I know the current class isn't a BaseStepCheckpoint, but I suggest that in our future BaseStepCheckpoint that we do `self.checkpoint_step(self)`. 

In light of this renaming information, I'd also suggest that the method `BaseDataCheckpoint.save_checkpoint` be renamed to `BaseDataCheckpoint.checkpoint_data` as `checkpoint` can be a verb and that we can have the `checkpoint_data` as well as the `checkpoint_step` version.

Reply:
EDIT: I saw that we have issue #50 already, I'll copy this comment there and resolve this comment.